pkg/lib/oauth/authz_service.go

```diff
@@ -57,13 +59,33 @@ func (s *AuthorizationService) Delete(a *Authorization) error {
 	// delete the offline grants that belong to the authorization
 	for _, sess := range sessions {
 		if offlineGrant, ok := sess.(*OfflineGrant); ok {
-			// TODO(DEV-1403): Check all authorization ids?
 			if offlineGrant.AuthorizationID == a.ID {
 				err := s.OAuthSessionManager.Delete(sess)
 				if err != nil {
 					return err
 				}
 			}
+			newTokens := []OfflineGrantRefreshToken{}
+			isTokenChanged := false
+			// Revoke any sub-tokens using this authorization
+			for _, token := range offlineGrant.RefreshTokens {
```

Comment:
Do not expose RefreshTokens

Reply:
`RefreshTokens` must be exposed, because the OfflineGrant is not contructed in the same package. However, I will remove usage of `RefreshTokens` outside the package.